ksql-engine/src/main/java/io/confluent/ksql/util/AvroUtil.java

```diff
@@ -0,0 +1,174 @@
+        abstractStreamCreateStatement = addAvroFields(abstractStreamCreateStatement, schema,
+                                                      schemaMetadata.getId());
+        return new Pair<>(abstractStreamCreateStatement, SqlFormatter.formatSql(abstractStreamCreateStatement));
+      } else {
+        return new Pair<>(abstractStreamCreateStatement, null);
+      }
+    } catch (Exception e) {
+      String errorMessage = String.format(" Could not fetch the AVRO schema from schema "
+                                          + "registry. %s ", e.getMessage());
+      throw new KsqlException(errorMessage);
+    }
+  }
+
+  private static AbstractStreamCreateStatement addAvroFields(AbstractStreamCreateStatement
+                                                                             abstractStreamCreateStatement, Schema
+                                                                             schema, int schemaId) {
+    List<TableElement> elements = new ArrayList<>();
+    for (Field field: schema.fields()) {
+      TableElement tableElement = new TableElement(field.name(), SchemaUtil
+          .getSQLTypeName(field.schema()));
+      elements.add(tableElement);
+    }
+    StringLiteral schemaIdLiteral = new StringLiteral(String.format("%d", schemaId));
+    Map<String, Expression> properties;
+    if (abstractStreamCreateStatement.getProperties().containsKey(KsqlConstants.AVRO_SCHEMA_ID)) {
+      properties = abstractStreamCreateStatement.getProperties();
+    } else {
+      properties = new HashMap<>();
+      properties.putAll(abstractStreamCreateStatement.getProperties());
+      properties.put(KsqlConstants.AVRO_SCHEMA_ID, schemaIdLiteral);
+    }
+
+
+    if (abstractStreamCreateStatement instanceof CreateStream) {
+      return new CreateStream(abstractStreamCreateStatement.getName(), elements,
+                              ((CreateStream) abstractStreamCreateStatement).isNotExists(),
+                              properties);
+    } else if (abstractStreamCreateStatement instanceof CreateTable) {
+      return new CreateTable(abstractStreamCreateStatement.getName(), elements,
+                             ((CreateStream) abstractStreamCreateStatement).isNotExists(),
+                             properties);
+    } else {
+      throw new KsqlException(String.format("Invalid AbstractStreamCreateStatement: %s",
+                                            SqlFormatter.formatSql(abstractStreamCreateStatement)));
+    }
+  }
+
+
+  public static synchronized void validatePersistantQueryResults(PersistentQueryMetadata
+                                                                     persistentQueryMetadata,
+                                                                 SchemaRegistryClient
+                                                                     schemaRegistryClient) {
+    if (persistentQueryMetadata.getResultTopic().getKsqlTopicSerDe().getSerDe() == DataSource.DataSourceSerDe.AVRO) {
```

Comment:
We should avoid having chained calls like this. There are multiple chances for an NPE to happen in that chain and it is exposing the inner details of the object structure. Would be better to add a method to `PersistentQueryMetadata` `getResultSerde()` and then delegate from there